### ISSUE TITLE
 org.tensorflow.TensorFlowException: Op type not registered 'Sum' in binary running on localhost. Make sure the Op and Kernel are registered in the binary running in this process.

### ISSUE THREAD

**Issue by eli99999** [tensorflow/tensorflow#14215]
### System information
- **Have I written custom code**:
I worte a code that is based on this : https://www.tensorflow.org/api_guides/python/contrib.signal
trying to produce mfcc works fine in python but i cannot make it work in Android

- **OS Platform and Distribution  Linux Ubuntu 16.04 
- **TensorFlow installed from source  **:
- **TensorFlow version 1.4.2:
- **Python version - 2.7 
- **Bazel version  0.7.0
-
 

### Describe the problem
i have reduced my graph to this - 
****PYTHON CODE that creates the pb *****
import sugartensor as tf
from tensorflow.python.framework import graph_util
from model import *
import data
import os
import sys
from tensorflow.contrib.framework.python.ops import audio_ops as contrib_audio
import functools

from tensorflow.contrib.session_bundle import exporter
from tensorflow.python.tools import freeze_graph
global session_config
exportfilename = 'out2/export_mfcc_only.pb'
with tf.device('/cpu:0'):
	session_config = tf.ConfigProto(allow_soft_placement=True, log_device_placement=False)
	 
	tf.app.flags.DEFINE_integer ('export_version',   1,           'version number of the exported model')
	session = tf.InteractiveSession()
	#session = tf.Session()
	# Run inference
	batch_size = 1
	voca_size = data.voca_size
	sample_rate = 16000.0	
	wavdata = tf.placeholder(tf.float32, [ None], name="wav_float_input")
	pcm = tf.expand_dims(wavdata, 0)
	stfts = tf.contrib.signal.stft(pcm, frame_length=2048, frame_step=512,
                             fft_length=2048,window_fn=functools.partial(tf.contrib.signal.hann_window, periodic=False), pad_end=True)
	spectrograms = tf.abs(stfts)
	# Warp the linear scale spectrograms into the mel-scale.
	num_spectrogram_bins = stfts.shape[-1].value
	lower_edge_hertz, upper_edge_hertz, num_mel_bins = 0.0,8000.0, 128
	linear_to_mel_weight_matrix = tf.contrib.signal.linear_to_mel_weight_matrix(
	num_mel_bins, num_spectrogram_bins, sample_rate, lower_edge_hertz,
	upper_edge_hertz)
	mel_spectrograms = tf.tensordot(
	spectrograms, linear_to_mel_weight_matrix, 1)
	mel_spectrograms.set_shape(spectrograms.shape[:-1].concatenate(
	linear_to_mel_weight_matrix.shape[-1:]))
	# Compute a stabilized log to get log-magnitude mel-scale spectrograms.
	log_mel_spectrograms = tf.log(mel_spectrograms + 1e-6)
	# Compute MFCCs from log_mel_spectrograms and take the first 13.
	mfccs = tf.contrib.signal.mfccs_from_log_mel_spectrograms(
	log_mel_spectrograms)[..., :20]

	seq_len =[tf.size(mfccs.sg_int().sg_sum(axis=2), name="output_node" )]

  
	

	session.run(tf.local_variables_initializer() )
	tf.global_variables_initializer()
	subgraph = tf.graph_util.extract_sub_graph(session.graph_def, ["output_node"])

	frozen_graph_def = graph_util.convert_variables_to_constants(
	      session, subgraph, ["output_node"])
	tf.train.write_graph(
	      frozen_graph_def,
	      os.path.dirname(exportfilename),
	      os.path.basename(exportfilename),
	      as_text=False)
	tf.logging.info('Saved frozen graph to %s',exportfilename)
***END OF PYTHON CODE***
 
i manage to load it and test the created pb it in python

i create libtensorflow_inference.so
using print_selective_registration_header as described in - 
tensorflow/tensorflow/python/tools/print_selective_registration_header.py

testing the code in Androids results in an Error when i load the model - 

11-03 16:15:31.790 8639-8639/org.tensorflow.demo E/native: executor.cc:651 Executor failed to create kernel. Not found: No registered 'ListDiff' OpKernel for CPU devices compatible with node Tensordot/ListDiff_1 = ListDiff[T=DT_INT32, out_idx=DT_INT32, _device="/device:CPU:0"](Tensordot/range_3, Tensordot/add_3)
                                                           	.  Registered:  <no registered kernels>
                                                           
                                                           	 [[Node: Tensordot/ListDiff_1 = ListDiff[T=DT_INT32, out_idx=DT_INT32, _device="/device:CPU:0"](Tensordot/range_3, Tensordot/add_3)]]
11-03 16:15:31.790 8639-8639/org.tensorflow.demo E/native: executor.cc:651 Executor failed to create kernel. Not found: No registered 'ListDiff' OpKernel for CPU devices compatible with node Tensordot/ListDiff_1 = ListDiff[T=DT_INT32, out_idx=DT_INT32, _device="/device:CPU:0"](Tensordot/range_3, Tensordot/add_3)
                                                           	.  Registered:  <no registered kernels>
                                                           
                                                           	 [[Node: Tensordot/ListDiff_1 = ListDiff[T=DT_INT32, out_idx=DT_INT32, _device="/device:CPU:0"](Tensordot/range_3, Tensordot/add_3)]]
11-03 16:15:31.800 8639-8639/org.tensorflow.demo E/native: executor.cc:651 Executor failed to create kernel. Not found: No registered 'ListDiff' OpKernel for CPU devices compatible with node Tensordot/ListDiff_1 = ListDiff[T=DT_INT32, out_idx=DT_INT32, _device="/device:CPU:0"](Tensordot/range_3, Tensordot/add_3)
                                                           	.  Registered:  <no registered kernels>
                                                           
                                                           	 [[Node: Tensordot/ListDiff_1 = ListDiff[T=DT_INT32, out_idx=DT_INT32, _device="/device:CPU:0"](Tensordot/range_3, Tensordot/add_3)]]
11-03 16:15:31.810 8639-8639/org.tensorflow.demo E/native: executor.cc:651 Executor failed to create kernel. Not found: No registered 'ListDiff' OpKernel for CPU devices compatible with node Tensordot/ListDiff = ListDiff[T=DT_INT32, out_idx=DT_INT32, _device="/device:CPU:0"](Tensordot/range_2, Tensordot/add_1)
                                                           	.  Registered:  <no registered kernels>
                                                           
                                                           	 [[Node: Tensordot/ListDiff = ListDiff[T=DT_INT32, out_idx=DT_INT32, _device="/device:CPU:0"](Tensordot/range_2, Tensordot/add_1)]]
11-03 16:15:31.870 8639-8639/org.tensorflow.demo E/AndroidRuntime: FATAL EXCEPTION: main
                                                                   Process: org.tensorflow.demo, PID: 8639
                                                                   org.tensorflow.TensorFlowException: Op type not registered 'Sum' in binary running on localhost. Make sure the Op and Kernel are registered in the binary running in this process.

i see that both 'ListDiff' and 'Sum' ops are in the ops_to_register.h file that is generated by the print_selective_registration_header process

is there a workaround for this?

**Comment by eli99999** [comment]
I looked into it, and it seems that there is a problem with the bazel compilation
i changed the models a few time and then did the print_selective_registration_header process,
each time the ops_to_register.h file changed but the libtensorflow_inference.so does not.
I tried all sort of bazel clean and bazel dump, and not chaned the resulted libtensorflow_inference.so
so i think that either there is some issue with the bazel caching or that the bug is in the print_selective_registration_header process

**Comment by mtu** [comment]
Hi, I met the same problem, do you sovle it at last?

**Comment by eli99999** [comment]
@mtu  -  eventually i had 2 ops_to_register.h in my project and bazel always compiled libtensorflow_inference.so according to the wrong one...

**Comment by warriorZH** [comment]
Hi @eli99999 i have generated ops_to_register.h by using tensorflow/python/tools/print_selective_registration_header.py. And then copy it into "tensorflow/core/framework/", and after bazel build the libtensorflow_inference.so also changed.
But my error log is still appear.

error log:
E/native: executor.cc:643 Executor failed to create kernel. Not found: No registered 'ListDiff' OpKernel for CPU devices compatible with node encoder/input_projection/Tensordot/ListDiff = ListDiff[T=DT_INT32, out_idx=DT_INT32](encoder/input_projection/Tensordot/range, encoder/input_projection/Tensordot/add_1).  Registered:  <no registered kernels>
          	 [[Node: encoder/input_projection/Tensordot/ListDiff = ListDiff[T=DT_INT32, out_idx=DT_INT32](encoder/input_projection/Tensordot/range, encoder/input_projection/Tensordot/add_1)]]

sys:
tf: 1.4.0  (I don't know where to get the 1.4.2 version)
bazel: 0.8.0

add: and  i find my error at "executor.cc:643" and yours at "executor.cc:651". If it's convenient, please let me know how to solve this problem. thanks

**Comment by jackyLens** [comment]
I met the same issue. Does anyone solved it?
